### ISSUE TITLE
Error-Can't connect to always on VPN

### ISSUE THREAD

**Issue by onlytanmoy** [IngoZenz/personaldnsfilter#156]
Hi, I am using the latest version of pDNSf from Playstore, quite often I am getting the below error even though Always-on VPN is enabled for pDNSf Service, any solution to this please?

![image](https://user-images.githubusercontent.com/48493659/118279285-e004d400-b4c2-11eb-88ac-2f340014fe6f.png)

**Comment by IngoZenz** [comment]
Thanks for letting me know.
Could you give the spec of your phone?
When you start manually, what folder is it using for the configuration?
It is logged at the beginning when starting the app.

**Comment by onlytanmoy** [comment]
@IngoZenz  I am using S20 FE 5G (https://www.gsmarena.com/samsung_galaxy_s20_fe_5g-10377.php)
Please see folder location in the screenshot below
![image](https://user-images.githubusercontent.com/48493659/118318026-b95d9200-b4f0-11eb-9fdf-aac25c07b2c1.png)

**Comment by IngoZenz** [comment]
Thanks for letting me know.
Could you give the spec of your phone?
When you start manually, what folder is it using for the configuration?
It is logged at the beginning when starting the app.

**Comment by onlytanmoy** [comment]
@IngoZenz I don't understand, I already gave you the details you asked for, right?

**Comment by IngoZenz** [comment]
Yes - all good - seems there was an issue with Github - I did not see the update of this issue - now it is fine...
Thanks for the information!

**Comment by onlytanmoy** [comment]
@IngoZenz I am on v1.50.48.1, facing issue in that.

**Comment by IngoZenz** [comment]
I think this is a bug with the new version related to storage access affe secting few devices and users.
[dnsfilter.1504802-dev.zip](https://github.com/IngoZenz/personaldnsfilter/files/6485339/dnsfilter.1504802-dev.zip)

You might help me to understand the issue. If you wish try with attached dev build and let me know if it solves the issue for you.
As this is not the playstore build, you'll need to uninstall the ps version first.

**Comment by onlytanmoy** [comment]
Thanks mate, I really appreciate that.
Forgot to clarify one thing, please see screenshot below, do I need to disable 'Optimise battery usage' for pDSNf? Not sure if that is what is causing the occasional drop outs in the connection.
![image](https://user-images.githubusercontent.com/48493659/118356903-45160380-b56f-11eb-9e65-3cb76335c610.png)

**Comment by IngoZenz** [comment]
So far I understood you face the issue after boot - but in case you face it "in between" indeed the battery optimization migtht cause this. It is better to have it disabled - it causes issues some some devices...
This would not be related to the new version but for older versions as well.
In PS error log I noticed some erros with accessing the storage since it was moved to new location in 1.50.48 - thought that is your issue as well ...

**Comment by onlytanmoy** [comment]
Oh no, not after boot, I faced it occasionally in between my phone usage.
I will disable the battery optimization for the app and see if that resolves it, my 2 cents it will.
Let this thread be open for now, I will close it after some days, hope that is fine with you.

**Comment by IngoZenz** [comment]
sure - that is fine ...
